panic on duplicate entries

When my list have a duplicate entries I got a panic and application closes:

[GIN-debug] GET    /movie/FIVEVOD0419/jL7FXAdJhl/5043.mp4 --> routes.github_com_pierre_emmanuelJ_iptv_proxy_pkg_routes..thunk11 (5 handlers)
[GIN-debug] GET    /movie/FIVEVOD0419/jL7FXAdJhl/15057.mp4 --> routes.github_com_pierre_emmanuelJ_iptv_proxy_pkg_routes..thunk11 (5 handlers)
[GIN-debug] GET    /movie/FIVEVOD0419/jL7FXAdJhl/3655.mp4 --> routes.github_com_pierre_emmanuelJ_iptv_proxy_pkg_routes..thunk11 (5 handlers)
[GIN-debug] GET    /movie/FIVEVOD0419/jL7FXAdJhl/3655.mp4 --> routes.github_com_pierre_emmanuelJ_iptv_proxy_pkg_routes..thunk11 (5 handlers)
panic: handlers are already registered for path '/movie/FIVEVOD0419/jL7FXAdJhl/3655.mp4'

goroutine 1 [running]:
panic
        ../../../src/libgo/go/runtime/panic.go:554
github_com_pierre_emmanuelJ_iptv_proxy_vendor_github_com_gin_gonic_gin.node.addRoute
        /root/go/src/github.com/pierre-emmanuelJ/iptv-proxy/vendor/github.com/gin-gonic/gin/tree.go:235
github_com_pierre_emmanuelJ_iptv_proxy_vendor_github_com_gin_gonic_gin.Engine.addRoute
        /root/go/src/github.com/pierre-emmanuelJ/iptv-proxy/vendor/github.com/gin-gonic/gin/gin.go:252
github_com_pierre_emmanuelJ_iptv_proxy_vendor_github_com_gin_gonic_gin.RouterGroup.handle
        /root/go/src/github.com/pierre-emmanuelJ/iptv-proxy/vendor/github.com/gin-gonic/gin/routergroup.go:71
github_com_pierre_emmanuelJ_iptv_proxy_vendor_github_com_gin_gonic_gin.RouterGroup.GET
        /root/go/src/github.com/pierre-emmanuelJ/iptv-proxy/vendor/github.com/gin-gonic/gin/routergroup.go:99
github_com_pierre_emmanuelJ_iptv_proxy_pkg_routes.Routes
        /root/go/src/github.com/pierre-emmanuelJ/iptv-proxy/pkg/routes/routes.go:75
github_com_pierre_emmanuelJ_iptv_proxy_pkg_routes.Serve
        /root/go/src/github.com/pierre-emmanuelJ/iptv-proxy/pkg/routes/routes.go:37
cmd.func1
        /root/go/src/github.com/pierre-emmanuelJ/iptv-proxy/cmd/root.go:52
github_com_pierre_emmanuelJ_iptv_proxy_vendor_github_com_spf13_cobra.Command.execute
        /root/go/src/github.com/pierre-emmanuelJ/iptv-proxy/vendor/github.com/spf13/cobra/command.go:766
github_com_pierre_emmanuelJ_iptv_proxy_vendor_github_com_spf13_cobra.Command.ExecuteC
        /root/go/src/github.com/pierre-emmanuelJ/iptv-proxy/vendor/github.com/spf13/cobra/command.go:852
github_com_pierre_emmanuelJ_iptv_proxy_vendor_github_com_spf13_cobra.Command.Execute
        /root/go/src/github.com/pierre-emmanuelJ/iptv-proxy/vendor/github.com/spf13/cobra/command.go:800
github_com_pierre_emmanuelJ_iptv_proxy_cmd.Execute
        /root/go/src/github.com/pierre-emmanuelJ/iptv-proxy/cmd/root.go:61
main.main
        /root/go/src/github.com/pierre-emmanuelJ/iptv-proxy/main.go:6

--- a/pkg/routes/routes.go
+++ b/pkg/routes/routes.go
@@ -84,6 +84,7 @@ func Routes(proxyConfig *config.ProxyConfig, r *gin.RouterGroup) {
 	}
 	p.newM3U = newM3U
 
+	checkList := map[string]int8{}
 	for i, track := range proxyConfig.Playlist.Tracks {
 		oriURL, err := url.Parse(track.URI)
 		if err != nil {
@@ -94,7 +95,15 @@ func Routes(proxyConfig *config.ProxyConfig, r *gin.RouterGroup) {
 			&proxyConfig.Playlist.Tracks[i],
 			nil,
 		}
+		_, ok := checkList[oriURL.Path]
+		if ok {
+			log.Printf("[iptv-proxy] WARNING endpoint %q already exist, skipping it", oriURL.Path)
+			continue
+		}
+
 		r.GET(oriURL.Path, p.authenticate, tmp.reverseProxy)
+
+		checkList[oriURL.Path] = 0
 	}
 }
 

